hibernate-core/src/main/java/org/hibernate/resource/jdbc/ResourceRegistry.java

```diff
@@ -56,6 +58,13 @@
 	void register(ResultSet resultSet, Statement statement);
 
```

Comment:
`ResourceRegistry` is for JDBC resources (e.g., `ResultSet`, `Statement`, `Blob`, `Clob`, `NClob`.

`Batch` is not a JDBC resource, so it seems strange to register it with a `ResourceRegistry`.